File: src/main/java/uk/laxd/dndSimulator/equipment/Greatsword.java
package uk.laxd.dndSimulator.equipment;

import uk.laxd.dndSimulator.ability.Ability;
import uk.laxd.dndSimulator.action.MeleeAttackAction;
import uk.laxd.dndSimulator.action.DamageType;
import uk.laxd.dndSimulator.dice.Die;

import java.util.Arrays;
import java.util.Collection;

public class Greatsword extends Weapon {
    @Override
    public int getToHitModifier(MeleeAttackAction attackAction) {
        return attackAction.getPerformer().getAbilityModifier(Ability.STRENGTH) + attackAction.getPerformer().getProficiencyBonus() + 1;
    }

    @Override
    public Collection<Die> getDamageDice(MeleeAttackAction attackAction) {
        return Arrays.asList(
                Die.D6,
                Die.D6
        );
    }

    @Override
    public int getDamage(MeleeAttackAction attackAction) {
        return attackAction.getPerformer().getAbilityModifier(Ability.STRENGTH);
    }

    @Override
    public Collection<WeaponProperty> getProperties() {
        return Arrays.asList(
                WeaponProperty.HEAVY
        );
    }

    @Override
    public DamageType getDamageType() {
        return DamageType.SLASHING;
    }
}

File: app/src/main/kotlin/uk/laxd/dndSimulator/equipment/Greatsword.kt
package uk.laxd.dndSimulator.equipment

import uk.laxd.dndSimulator.action.MeleeAttackAction
import uk.laxd.dndSimulator.ability.Ability
import uk.laxd.dndSimulator.dice.Die
import uk.laxd.dndSimulator.action.DamageType

class Greatsword : Weapon("Greatsword") {
    override var toHitModifier = 1

    override fun getDamageDice(attackAction: MeleeAttackAction): MutableList<Die> {
        return mutableListOf(
                Die.D6,
                Die.D6
        )
    }

    override fun getDamage(attackAction: MeleeAttackAction): Int {
        return attackAction.actor.getAbilityModifier(Ability.STRENGTH)
    }

    override val properties: List<WeaponProperty>
        get() = listOf(
                WeaponProperty.HEAVY
        )

    override val damageType = DamageType.SLASHING
    override val range = 5
    override val priority = 0.8
    override fun getProficiencyNames(): List<String> = listOf("martial-weapons", "greatsword")
}